packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -48,6 +49,34 @@ export class AssociationManager {
     }
   }
 
+  public static associateTagOptions(portfolio: IPortfolio, tagOptions: TagOptions): void {
+    Object.keys(tagOptions.tagOptionsMap).forEach(key => {
+      InputValidator.validateLength(portfolio.node.addr, 'TagOption key', 1, 128, key);
+      tagOptions.tagOptionsMap[key].forEach((value: string) => {
+        InputValidator.validateLength(portfolio.node.addr, 'TagOption value', 1, 256, value);
+        const stack = cdk.Stack.of(portfolio);
```

Comment:
Let's rename this variable to `portfolioStack`.

Reply:
Also, let's move this out of the loop (it should be done once per `associateTagOptions()` call, not on each loop iteration).